tensorboard/plugins/graph/graph_util_test.py

```diff
@@ -214,7 +214,84 @@ def test_combine_graph_defs_name_collided_different_content(self):
     with six.assertRaisesRegex(
         self,
         ValueError,
-        r'Cannot combine GraphDefs because nodes share a name but are different. X'):
+        ('Cannot combine GraphDefs because nodes share a name but '
+         'contents are different: X')):
+      graph_util.combine_graph_defs(graph_def_a, graph_def_b)
+
+  def test_combine_graph_defs_dst_nodes_keys_non_unique(self):
+    graph_def_a = GraphDef()
+    text_format.Merge('''
+      node {
+        name: "X"
+        op: "Input"
+      }
+      node {
+        name: "X"
+        op: "Input"
+      }
+      versions {
+        producer: 21
+      }
+    ''', graph_def_a)
+
+    graph_def_b = GraphDef()
+    text_format.Merge('''
+      node {
+        name: "X"
+        op: "Input"
+        device: "cpu:0"
+      }
+      node {
+        name: "Z"
+        op: "Input"
+      }
+      versions {
+        producer: 21
+      }
+    ''', graph_def_b)
+
+    with six.assertRaisesRegex(
+        self,
+        ValueError,
+        ('A GraphDef contains non-unique node names: X')):
```

Comment:
The inputs in this test case are problematic for two reasons: the first
graph has multiple nodes called “X”, and the second graph has a node “X”
that is different from the “X” nodes in the first graph. It seems to me
that the `combine_graph_defs` function would be within its rights to
raise _either_ of the relevant exceptions. Can we amend this test so
that the inputs have exactly one problem? (e.g., remove “X” from the
second graph)